picard/ui/scripteditor.py

```diff
@@ -0,0 +1,428 @@
+            self.ui.file_naming_format.setPlainText(str(script_text).strip())
+
+    def show_scripting_documentation(self):
+        ScriptingDocumentationDialog.show_instance(parent=self)
+
+    def check_formats(self):
+        self.test()
+        self.update_examples()
+
+    def update_example_files(self):
+        self.examples.update_sample_example_files()
+        self.update_examples()
+
+    def update_examples(self, override=None, send_signal=True):
+        if not override or not isinstance(override, dict):
+            override = self.override
+        else:
+            self.override = override
+        override['file_naming_format'] = self.ui.file_naming_format.toPlainText()
+
+        self.ui.example_filename_before.clear()
+        self.ui.example_filename_after.clear()
+        self.current_row = -1
+
+        examples = self.examples.get_examples(override)
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            self.ui.example_filename_before.addItem(before)
+            self.ui.example_filename_after.addItem(after)
+
+        if send_signal:
+            self.signal_update.emit()
+
+    def toggle_wordwrap(self):
+        if self.ui.file_naming_word_wrap.isChecked():
+            self.wordwrap = QtWidgets.QTextEdit.WidgetWidth
+        else:
+            self.wordwrap = QtWidgets.QTextEdit.NoWrap
+        self.ui.file_naming_format.setLineWrapMode(self.wordwrap)
+
+    def import_script(self):
+        options = QtWidgets.QFileDialog.Options()
+        options |= QtWidgets.QFileDialog.DontUseNativeDialog
+        filename, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Import Script File", "", "All Files (*);;Picard Script Files (*.pts)", options=options)
+        if filename:
+            log.debug('Importing naming script file: %s' % filename)
+            file_text = ""
+            try:
+                with open(filename, 'r', encoding='utf8') as i_file:
+                    file_text = i_file.read()
+            except OSError as error:
+                self.display_error(error)
+                return
+            if file_text.strip():
+                self.ui.file_naming_format.setPlainText(file_text.strip())
+            else:
+                self.ui.file_naming_format.setPlainText("$noop( The import file was empty. )")
+
+    def export_script(self):
+        script_text = self.get_script()
+        if script_text:
+            options = QtWidgets.QFileDialog.Options()
+            options |= QtWidgets.QFileDialog.DontUseNativeDialog
+            filename, _ = QtWidgets.QFileDialog.getSaveFileName(self, "Export Script File", "picard_naming_script.pts", "All Files (*);;Picard Script Files (*.pts)", options=options)
+            if filename:
+                log.debug('Exporting naming script file: %s' % filename)
+                try:
+                    with open(filename, 'w', encoding='utf8') as o_file:
+                        o_file.write(self.get_script() + '\n')
```

Comment:
We can do that, but I was thinking more about someone saving the text file for posting to github (such as https://github.com/rdswift/picard-scripts/blob/master/naming_scripts/Bob_Swift/Bob%20Swift's%20Naming%20Script.pts) or if they wanted to use an external editor.

I assumed that the script name / script text combination would be saved to the user configuration settings, similar to what's done with the tagging scripts (and that this would follow in a later PR).

Reply:
Ok, to KISS, a simple text file is good for sharing. I misunderstood the purpose.
But I was also thinking about saving scripts "configuration" to separate files, so we could provide those as a service on the website (a repository of filenaming scripts).
That's too early. So let's stick to the simple (and good) approach for now.